pkg/bundle/read.go

```diff
@@ -247,6 +250,39 @@ func read(ctx context.Context, name, baseDir string, bundleSpecReader io.Reader,
 	return New(def, scans...)
 }
 
+// fixAndApplyHelmSpec checks if the spec's Helm repo, if there is one, ends with a '/' and appends one if not. See chartURL
+// This function also propagates root Helm chart properties to the child targets and performs the repo check.
+// Note: The repo and version properties of child targets are only propagated if either property is overridden. This
+// allows child targets to specify a path to a chart, to specify a url to the chart(e.g. oci://), etc.
+func fixAndApplyHelmSpec(spec *fleet.BundleSpec) {
+	if spec.Helm != nil && spec.Helm.Chart != "" {
+		if spec.Helm.Repo != "" && !strings.HasSuffix(spec.Helm.Repo, "/") {
+			spec.Helm.Repo = spec.Helm.Repo + "/"
+		}
+	}
+
+	for _, target := range spec.Targets {
+		if target.Helm != nil && spec.Helm != nil {
+			if target.Helm.Repo != "" || target.Helm.Version != "" {
```

Comment:
I think this block of code was trying to handle a use case like the first target customization below:
```
namespace: fleet-mc-helm-external-example
helm:
  repo:  https://github.com/rancher/fleet-examples/releases/download/example/
  chart: guestbook
  version: 0.0.0
targetCustomizations:
- name: dev
  helm:
    chart: "./guestbook-dev/"
  clusterSelector:
    matchLabels:
      env: dev

- name: test
  helm:
    chart: oci://docker.github.com/rancher/fleet-examples/guestbook
    version: 0.1.0-test
  clusterSelector:
    matchLabels:
      env: test
```
I'm not sure if the dev customization is a valid, but I think that's what I was trying to account for.

On second look though, I am noticing that the code above will cause issues with the second target customization. This `if` and the code in `calculate.go` will generate a `HelmOptions` like the below:
```
helm:
  repo:  https://github.com/rancher/fleet-examples/releases/download/example/
  chart: oci://docker.github.com/rancher/fleet-examples/guestbook
  version: 0.1.0-test
```
I'm not entirely sure how this will be handled.

Reply:
The issue I was trying to address was this: If the user changes the Helm "reference" type(standard repository, OCI repository, local path chart, URI) in a customization, how should options be merged(because a straight merge may override too much and cause issues). I'm not sure but doing the merge of options in `calculate.go` may not provide enough content.

I think I have a way to handle this better than how I did here and I'll implement it later today.

Reply:
Personally I would prefer no magic. If the user wants defaults to be overwritten, s/he should do so in the `targetCustomizations` for all preferences they want to change, because they know what they want to achieve and what the default is.
If Fleet has some logic to reset some of the defaults and in other cases not it would be very confusing.